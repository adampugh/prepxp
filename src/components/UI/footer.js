import React from "react";
import { Link } from "react-router-dom";

const footer = () => (
    <footer>
        <div className="container">
            <div className="footer__grid">
                <div className="footer__grid__panel">
                    <h1 className="logo">PrepXP</h1>
                    <p>
                        <i className="fab fa-twitter"></i>
                        <i className="fab fa-facebook-f"></i>
                        <i className="fab fa-youtube"></i>    
                    </p>
                    <p>info@prepxp.com</p>
                </div>
                <div className="footer__grid__panel">
                    <h2>Explore</h2>
                    <Link to="/prepxp/dashboard"><p className="footer__grid__panel__link">•  Home</p></Link>
                    <Link to="/prepxp/"><p className="footer__grid__panel__link">•  About</p></Link>
                    <Link to="/prepxp/blog"><p className="footer__grid__panel__link">•  Blog</p></Link>
                </div>
                <div className="footer__grid__panel">
                    <h2>Latest Blog Posts</h2>
                    <div className="footer__grid__blogPost">
                        <Link to="/prepxp/blog"><p>5 Most common interview questions</p></Link>
                    </div>
                    <div className="footer__grid__blogPost">
                        <Link to="/prepxp/blog"><p>How to pass the technical interview</p></Link>
                    </div>
                </div>
            </div>
            <p className="footer__tagline">© 2017 Design By <a href="https://adampugh.github.io/kokonoka/" rel="noopener noreferrer" target="_blank">Kokonoka Digital</a></p>
        </div>
    </footer>
);

export default footer;