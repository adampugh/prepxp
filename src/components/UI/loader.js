import React from "react";

const loader = () => (
    <div className="loader__container">
        <div className="loader">Loading...</div>
    </div>
);

export default loader;