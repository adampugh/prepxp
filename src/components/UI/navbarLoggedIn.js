import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

import * as authActions from "../../store/actions/auth";

export class Navbar extends Component {
    state = {
        drawerOpen: false,
        transparentBg: false
    }

    componentDidMount() {
        document.addEventListener('scroll', () => {
            const transparentBg = window.scrollY > 81
            if (transparentBg !== this.state.transparentBg) {
                this.setState({ transparentBg });
            }
        });
    }

    openDrawer = () => {
        this.setState({
            drawerOpen: !this.state.drawerOpen,
        });
    }

    


    render() {
        return (
            <div>
                <div id="drawer" className={this.state.drawerOpen ? 'slideIn' : 'slideOut'}>
                    <Link to="/prexp/dashboard"><h1>Dashboard</h1></Link>
                    <Link to="/prexp/search"><h1>Search</h1></Link>
                    <Link to="/prexp/blog"><h1>Blog</h1></Link>
                    <h1 onClick={this.props.startLogout}>Logout</h1>
                    <p>
                        <i className="fab fa-twitter"></i>
                        <i className="fab fa-facebook-f"></i>
                        <i className="fab fa-youtube"></i>
                    </p>
                </div>
                <div className={`navbar__loggedIn ${ this.state.transparentBg ? "navbar__loggedIn--transparent" : ""}`}>
                    <div className="container-l"> 
                        <div className="navbar__loggedIn__div--mobile">
                            <h1 className="logo">PrepXP</h1>
                            <h1 onClick={this.openDrawer}>☰</h1>
                        </div>
                        <div className="navbar__loggedIn__div--desktop">
                            <h1 className="logo">PrepXP</h1>
                            <ul>
                                <Link to="/prexp/dashboard"><li>Dashboard</li></Link>
                                <span>•</span>
                                <Link to="/prexp/search"><li>Search</li></Link>
                                <span>•</span>
                                <Link to="/prexp/blog"><li>Blog</li></Link>
                            </ul>
                            <button onClick={this.props.startLogout} className="btn btn--black"><i className="fas fa-sign-out-alt"></i> Logout</button>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
};

const mapDispatchToProps = dispatch => {
    return {
        startLogout: () => dispatch(authActions.startLogout())
    }
}


export default connect(null, mapDispatchToProps)(Navbar);