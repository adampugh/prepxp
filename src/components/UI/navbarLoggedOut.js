import React, { Component } from "react";
import { Link } from "react-router-dom";

import * as mobile from "is-mobile";

// add modal in ui for menu - might need loggedin / not logged in versions

class Navbar extends Component {
    state = {
        isOnWhite: false,
        onMobile: false,
        drawerOpen: false,
        hamburgerIconWhite: false
    }
    
    componentDidMount() {
        const isMobile = mobile();
        this.setState({
            onMobile: isMobile
        });

        if (isMobile === false) {
            document.addEventListener('scroll', () => {
                const isOnWhite = window.scrollY > 1381 && window.scrollY < 2650;
                if (isOnWhite !== this.state.isOnWhite) {
                    this.setState({ isOnWhite });
                }
            });
        } else {
            document.addEventListener('scroll', () => {
                const isOnWhite = window.scrollY > 2000 && window.scrollY < 4500;
                if (isOnWhite !== this.state.isOnWhite) {
                    this.setState({ isOnWhite });
                }
            });
        }
    }

    openDrawer = () => {
        this.setState({
            drawerOpen: !this.state.drawerOpen,
            hamburgerIconWhite: !this.state.hamburgerIconWhite
        });
    }

    render() {
        return (
            <div>

                <div id="drawer" className={this.state.drawerOpen ? 'slideIn' : 'slideOut'}>
                    <h1>Home</h1>
                    <h1>About</h1>
                    <h1>Blog</h1>
                    <Link to="/login"><h1>Login</h1></Link>
                    <p>
                        <i className="fab fa-twitter"></i>
                        <i className="fab fa-facebook-f"></i>
                        <i className="fab fa-youtube"></i>
                    </p>
                </div>
                <div className="navbar">
                    <div className="container-l"> 
                        <div className={this.state.isOnWhite ? "navbar__loggedOut navbar__loggedOut--black" : "navbar__loggedOut"}>
                            <h1 className={this.state.onMobile && this.state.hamburgerIconWhite ? "logo navbar__white" : "logo"}>PrepXP</h1>
                            <h1 onClick={this.openDrawer} 
                                className={this.state.hamburgerIconWhite ? "navbar__white" : ""}>
                                â˜°
                            </h1>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default Navbar;