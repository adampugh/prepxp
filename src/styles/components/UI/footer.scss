footer {
    text-align: left;
    color: white;
    background: $black-gradient, url("../assets/images/bg3.jpg") no-repeat center center;
    background-size: cover;
    padding: $padding-s 0;
    .logo {
        margin-bottom: 0;
        margin-top: 15px;
    }
}

.footer__grid {
    display: grid;
    grid-template-columns: 1fr;
    width: 80%;
    margin: 0 auto;
    p {
        color: $light-grey;
    }
    i, svg {
        margin-right: 20px;
        cursor: pointer;
        &:hover {
            color: white;
        }
    }
}

.footer__grid__blogPost {
    border: 1px solid $light-grey;
    color: $light-grey;
    padding: 0 5px;
    margin: 5px 0;
    cursor: pointer;
    &:hover {
        border: 1px solid white;
        p {
            color: white;
        }
    }
    p {
        margin: 10px;
    }
}

.footer__grid__panel__link {
    cursor: pointer;
    &:hover {
        color: white;
    }
}

.footer__tagline {
    text-align: center;
    font-size: $font-s;
    padding-top: $padding-s;
    a {
        text-decoration: none;
        color: $off-purple;
    }
}

@media (min-width: $desktop-breakpoint) {
    .footer__grid {
        grid-template-columns: 1fr 1fr 1fr;
        width: 100%;
    }

    .footer__grid__panel {
        margin: 0 auto;
    }
}