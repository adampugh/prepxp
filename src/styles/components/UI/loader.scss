.loader__container {
    background: $very-light-grey;
    padding: $padding-l 0;
    box-sizing: border-box;
    height: 100vh;
    display: grid;
}

.loader,
.loader:after {
    border-radius: 50%;
    width: 10em;
    height: 10em;
}
.loader {
    align-self: center;
    margin: 60px auto;
    font-size: 10px;
    position: relative;
    text-indent: -19em;
    color: transparent;
    border-top: 1.1em solid rgba(172,102,196, 0.2);
    border-right: 1.1em solid rgba(172,102,196, 0.2);
    border-bottom: 1.1em solid rgba(172,102,196, 0.2);
    border-left: 1.1em solid #ac66c4;
    -webkit-transform: translateZ(0);
    -ms-transform: translateZ(0);
    transform: translateZ(0);
    -webkit-animation: load8 1.1s infinite linear;
    animation: load8 1.1s infinite linear;
}
@-webkit-keyframes load8 {
    0% {
        -webkit-transform: rotate(0deg);
        transform: rotate(0deg);
    }
    100% {
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}
@keyframes load8 {
    0% {
        -webkit-transform: rotate(0deg);
        transform: rotate(0deg);
    }
    100% {
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}