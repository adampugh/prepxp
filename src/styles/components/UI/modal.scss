.ReactModalPortal > div {
    opacity: 0;
}

.ReactModalPortal .ReactModal__Overlay {
    align-items: center;
    display: flex;
    justify-content: center;
    transition: opacity 200ms ease-in-out;
}

.ReactModalPortal .ReactModal__Overlay--after-open {
    opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
    opacity: 0;
}

.modal {
    color: $off-grey;
    background: white;
    border-radius: 20px;
    z-index: 2000;
    max-width: 80%;
    min-width: 10rem;
    // min-height: 18rem;
    outline: none;
    padding: 30px;
    text-align: center;
    position: absolute;
    h3 {
        display: inline;
    }
    hr {
        width: 80%;
        margin-bottom: 25px;
        color: lightgrey;
        background: lightgrey;
        border: 1px solid lightgrey;
        margin-top: 20px;
    }
    span {
        position: absolute;
        top: 14px;
        right: 20px;
        i, svg {
            color: $light-grey;
            cursor: pointer;
        }
    }
    input {
        display: block;
        width: 80%;
        margin: $padding-s auto;
        padding: 10px;
        border-radius: 5px;
        border: 1px solid $light-grey;
    }
}

.modal__error {
    color: lightcoral;
    padding: 0;
    margin: 0;
}

.modal--delete {
    button {
        margin: 0 10px;
    }
    .loader__container {
        height: 100%;
        display: block;
        background: white;
        padding: 0;
    }
    
    .loader,
    .loader:after {
        border-radius: 50%;
        width: 50px;
        height: 50px;
    }
    
}

.modal__icon {
    font-size: $font-l;
    margin: 0;
    color: lightgreen;
}

@media (min-width: $desktop-breakpoint) {
    .modal {
        span {
            i, svg {
                &:hover {
                    color: $off-grey;
                }
            }
        }
    }
}