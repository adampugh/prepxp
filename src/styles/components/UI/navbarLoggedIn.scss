.navbar__loggedIn {
    background: $black-gradient, url("../assets/images/bg5.jpg") no-repeat center center;
    color: white;
    position: fixed;
    width: 100%;
    z-index: 100000000;
    opacity: 1;
    transition: opacity .25s ease-in-out;
    -moz-transition: opacity .25s ease-in-out;
    -webkit-transition: opacity .25s ease-in-out;
    h1 {
        display: inline;
        cursor: pointer;
        z-index: 400;
    }
}

.navbar__loggedIn__div--desktop {
    display: none;
}

.navbar__loggedIn__div--mobile {
    display: flex;
    justify-content: space-between;
}

@media (min-width: $desktop-breakpoint) {
    .navbar__loggedIn__div--desktop {
        display: flex;
        justify-content: space-between;
        padding: 0;
        ul {
            list-style-type: none;
            padding-top: 12px;
        }
        li {
            display: inline;
            margin: 0 22px;
        }
        button {
            margin: 15px 0;
        }
        .btn--black {
            background: transparent;
            color: white;
        }



        a {
            position: relative;
            margin: 0 1em;
          }
          
          a:before {
            content: "";
            position: absolute;
            width: 20px;
            height: 2px;
            bottom: -5px;
            left: 40%;
            background-color: $off-purple;
            visibility: hidden;
            -webkit-transform: scaleX(0);
            transform: scaleX(0);
            -webkit-transition: all 0.3s ease-in-out 0s;
            transition: all 0.3s ease-in-out 0s;
          }
          
          a:hover:before {
            visibility: visible;
            -webkit-transform: scaleX(1);
            transform: scaleX(1);
          }
    }

    $animate: all 0.2s ease-in-out;

    .navbar__loggedIn--transparent {
        // background: rgba(0,0,0,0.7);
        opacity: 0.7;
    }


    .navbar__loggedIn__div--mobile {
        display: none;
    }
}