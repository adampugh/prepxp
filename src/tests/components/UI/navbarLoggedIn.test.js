import React from "react";
import { shallow } from "enzyme";

import { Navbar } from "../../../components/UI/navbarLoggedIn";

describe("<NavbarLoggedIn />", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(<Navbar />);
        expect(wrapper).toMatchSnapshot();
    });
});