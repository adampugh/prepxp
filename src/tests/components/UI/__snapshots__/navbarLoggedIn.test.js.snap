// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarLoggedIn /> should match snapshot 1`] = `
<div>
  <div
    className="slideOut"
    id="drawer"
  >
    <Link
      replace={false}
      to="/dashboard"
    >
      <h1>
        Dashboard
      </h1>
    </Link>
    <h1>
      Lists
    </h1>
    <h1>
      Blog
    </h1>
    <h1>
      Logout
    </h1>
    <p>
      <i
        className="fab fa-twitter"
      />
      <i
        className="fab fa-facebook-f"
      />
      <i
        className="fab fa-youtube"
      />
    </p>
  </div>
  <div
    className="navbar__loggedIn"
  >
    <div
      className="container-l"
    >
      <div
        className="navbar__loggedIn__div"
      >
        <h1
          className="logo"
        >
          PrepXP
        </h1>
        <h1
          onClick={[Function]}
        >
          â˜°
        </h1>
      </div>
    </div>
  </div>
</div>
`;
